core/auth/payment_metadata.go

```diff
@@ -0,0 +1,169 @@
+package auth
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+)
+
+/* SUBJECT TO MODIFICATIONS */
+
+// EIP712Signer handles EIP-712 domain specific signing operations over typed and structured data
+type EIP712Signer struct {
+	domain apitypes.TypedDataDomain
+	types  apitypes.Types
+}
+
+// NewEIP712Signer creates a new EIP712Signer instance
+func NewEIP712Signer(chainID *big.Int, verifyingContract common.Address) *EIP712Signer {
+	return &EIP712Signer{
+		domain: apitypes.TypedDataDomain{
+			Name:              "EigenDA",
+			Version:           "1",
+			ChainId:           (*math.HexOrDecimal256)(chainID),
+			VerifyingContract: verifyingContract.Hex(),
+		},
+		types: apitypes.Types{
+			"EIP712Domain": []apitypes.Type{
+				{Name: "name", Type: "string"},
+				{Name: "version", Type: "string"},
+				{Name: "chainId", Type: "uint256"},
+				{Name: "verifyingContract", Type: "address"},
+			},
+			"PaymentMetadata": []apitypes.Type{
```

Comment:
Does this depend on the structure of `BlobHeader` in v2? (i.e. would signature over `BlobHeader` be sufficient without one over `PaymentMetadata`?)
Do we need this struct signed in both v1 & v2?

Reply:
I think payment metadata will depend on `BlobHeader` v2 in which `PaymentMetadata` includes `BlobHeader`'s hash, a field like `BlobKey`. 
This struct should always be signed, but I guess payments is currently disjointed from blob information. We could discuss whether it is okay to let payment be "blob-agnostic" in v1?

Reply:
In order to support both v1 and v2, what if we were to add a `BlobKey` field here which is not populated in v1?